entrypoint.sh

```diff
@@ -39,7 +39,18 @@ fi
 # Run octodns-sync.
 echo "INFO: _config_path: ${_config_path}"
 if [ "${_doit}" = "--doit" ]; then
-  octodns-sync --config-file="${_config_path}" --doit
+  RESULT=$(octodns-sync --config-file="${_config_path}" --log-stream-stdout --doit)
 else
-  octodns-sync --config-file="${_config_path}"
+  RESULT=$(octodns-sync --config-file="${_config_path}" --log-stream-stdout)
 fi
+echo "$RESULT"
```

Comment:
Here, rather than route the output through an environment variable will it work to store it in a file? For example:

```
octodns-sync --config-file="${_config_path}" --log-stream-stdout | tee /tmp/octodns-sync.log
```

Then you could use the [_Setting the comment body from a file_](https://github.com/peter-evans/create-or-update-comment/tree/1a85b6614033a04e3a82c1d567d228facceb805c#setting-the-comment-body-from-a-file) approach to use the output in the next step of your job.

This way `octodns-sync` would still output directly for the Actions log and the unmodified output would remain available for other uses in addition to this specific case with **peter-evans**' Actions.

I'll setup a test for this later today or tomorrow. If you happen to check it in the meantime, @thattommyhall, I'll be curious to hear what you find.

Reply:
🎉 

@thattommyhall to illustrate that ☝️ I made an example in https://github.com/solvaholic/ideal-octo-pancake.

I pushed [a new version](https://github.com/solvaholic/octodns-sync/blob/0ae71779c9c0221504ac5bac27a649465b035a69/entrypoint.sh#L39) of the solvaholic/octodns-sync container to Docker hub that writes the octodns-sync output to a file:

```
# Run octodns-sync.
echo "INFO: _config_path: ${_config_path}"
if [ "${_doit}" = "--doit" ]; then
  octodns-sync --config-file="${_config_path}" \
  --log-stream-stdout --doit | tee ${GITHUB_WORKSPACE}/octodns-sync.log
else
  octodns-sync --config-file="${_config_path}" \
  --log-stream-stdout | tee ${GITHUB_WORKSPACE}/octodns-sync.log
fi
```

And added a step to the workflow job to parse the output from the file:

```
      - id: get-comment-body
        run: |
          body=$(cat ${GITHUB_WORKSPACE}/octodns-sync.log)
          body="${body//'%'/'%25'}"
          body="${body//$'\n'/'%0A'}"
          body="${body//$'\r'/'%0D'}" 
          echo ::set-output name=body::$body
```

The automation added this: https://github.com/solvaholic/ideal-octo-pancake/pull/1#issuecomment-706599506

Would that approach meet your needs?